runtime/cudaq/qis/qubit_qis.h

```diff
@@ -293,15 +525,44 @@ void swap(QubitArgs &...args) {
   getExecutionManager()->apply("swap", {}, controls, targets);
 }
 
-#if CUDAQ_USE_STD20
 template <typename QuantumRegister>
   requires(std::ranges::range<QuantumRegister>)
-#else
+void swap(QuantumRegister &ctrls, qubit &src, qubit &target) {
+  std::vector<QuditInfo> controls;
+  std::transform(ctrls.begin(), ctrls.end(), std::back_inserter(controls),
+                 [](const auto &q) { return qubitToQuditInfo(q); });
+  getExecutionManager()->apply(
+      "swap", {}, controls, {qubitToQuditInfo(src), qubitToQuditInfo(target)});
+}
+
+#else // not C++20
+
+template <typename Qubit>
+void swap(Qubit &src, Qubit &target) {
+  static_assert(std::is_same<qubit, Qubit>::value,
+                "Cannot operate on a qudit with Levels != 2");
+  std::vector<QuditInfo> qubitIds{qubitToQuditInfo(src),
+                                  qubitToQuditInfo(target)};
+  getExecutionManager()->apply("swap", {}, {}, qubitIds);
+}
+
+template <typename... QubitArgs>
+void cswap(QubitArgs &...args, qubit &src, qubit &target) {
```

Comment:
This is also a multi-control swap when cswap the name implies a singly-controlled swap

Reply:
This one didn't use the template. I'll remove that.